src/app/clusters/icd-management-server/icd-management-cluster.h

```diff
@@ -0,0 +1,171 @@
+/**
+ *
+ *    Copyright (c) 2022-25 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/icd/server/ICDConfigurationData.h>
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <app/server-cluster/OptionalAttributeSet.h>
+#include <app/util/basic-types.h>
+#include <clusters/IcdManagement/ClusterId.h>
+#include <clusters/IcdManagement/Commands.h>
+#include <clusters/IcdManagement/Metadata.h>
+#include <crypto/SessionKeystore.h>
+#include <lib/core/Optional.h>
+#include <lib/support/Span.h>
+
+#include <app/icd/server/ICDServerConfig.h>
+#include <app/server/Server.h>
+
+#if CHIP_CONFIG_ENABLE_ICD_CIP
+#include <app/icd/server/ICDMonitoringTable.h> // nogncheck
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#endif // CHIP_CONFIG_ENABLE_ICD_CIP
+
+namespace chip {
+namespace Crypto {
+using SymmetricKeystore = SessionKeystore;
+} // namespace Crypto
+
+namespace app {
+namespace Clusters {
+
+enum class OptionalCommands : uint32_t
+{
+    kStayActiveRequest  = 0x01,
+    kStayActiveResponse = 0x02,
```

Comment:
There is only one optional command.

This should not be under the global namespace, it should be under the ICD Cluster